pimd/pim6_mld.c

```diff
@@ -2737,8 +2737,11 @@ static void gm_show_joins_one(struct vty *vty, struct gm_if *gm_ifp,
 		}
 
 		js_src = json_object_new_object();
-		json_object_object_addf(js_group, js_src, "%pPA",
-					&sg->sgaddr.src);
+		if (pim_addr_is_any(sg->sgaddr.src))
+			json_object_object_add(js_group, "*", js_src);
+		else
+			json_object_object_addf(js_group, js_src, "%pPA",
```

Comment:
please use "pPAs" here instead of if else. printfrr_pimaddr api will take care of printing * if we pass 's' here.